### PR TITLE
DI-test-autowiring

### PR DESCRIPTION
**Pull request by gilles-g**

### CODE REVIEW

**Comment by gilles-g** [comment]
@chalasr If you have some times, did you know how to fix that test? :thinking: 
https://github.com/lexik/LexikCronFileGeneratorBundle/blob/master/Tests/DependencyInjection/LexikCronFileGeneratorExtensionTest.php

Help to debug, trace:

```bash
1) Lexik\Bundle\CronFileGeneratorBundle\Tests\DependencyInjection\LexikCronFileGeneratorExtensionTest::testLoadEmptyConfiguration
Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: You have requested a non-existent service "twig.loader".
```

**Comment by chalasr** [comment]
@spike31 here you go:
```diff
diff --git a/Tests/DependencyInjection/LexikCronFileGeneratorExtensionTest.php b/Tests/DependencyInjection/LexikCronFileGeneratorExtensionTest.php
index 10929db..05b4d5a 100644
--- a/Tests/DependencyInjection/LexikCronFileGeneratorExtensionTest.php
+++ b/Tests/DependencyInjection/LexikCronFileGeneratorExtensionTest.php
@@ -11,9 +11,11 @@ use Symfony\Bundle\FrameworkBundle\DependencyInjection\FrameworkExtension;
 use Symfony\Bundle\FrameworkBundle\FrameworkBundle;
 use Symfony\Bundle\TwigBundle\DependencyInjection\TwigExtension;
 use Symfony\Bundle\TwigBundle\TwigBundle;
+use Symfony\Component\Config\Loader\LoaderInterface;
 use Symfony\Component\DependencyInjection\Compiler\PassConfig;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\DependencyInjection\ParameterBag\ParameterBag;
+use Symfony\Component\HttpKernel\Kernel;
 
 class LexikCronFileGeneratorExtensionTest extends TestCase
 {
@@ -60,8 +62,7 @@ class LexikCronFileGeneratorExtensionTest extends TestCase
             ->setAutowired(true);
 
         $this->compileContainer($container);
-dd($container->get('templating'));
-        $autowired = $container->get('autowired');
+        $this->assertInstanceOf(Autowired::class, $container->get('autowired'));
     }
 
     private function createContainer(array $configs = [])
@@ -82,6 +83,14 @@ dd($container->get('templating'));
             ],
         ]));
 
+        $container->set('kernel', new class ('test', false) extends Kernel {
+            public function registerBundles()
+            {
+            }
+            public function registerContainerConfiguration(LoaderInterface $loader)
+            {
+            }
+        });
         $container->registerExtension(new FrameworkExtension());
         $container->registerExtension(new TwigExtension());
         $container->registerExtension(new LexikCronFileGeneratorExtension());
@@ -96,8 +105,10 @@ dd($container->get('templating'));
     private function compileContainer(ContainerBuilder $container)
     {
         $container->addCompilerPass(new TemplatingPass(), PassConfig::TYPE_BEFORE_REMOVING);
-        $container->getCompilerPassConfig()->setOptimizationPasses([]);
         $container->getCompilerPassConfig()->setRemovingPasses([]);
+
+        (new TwigBundle())->build($container);
+
         $container->compile();
     }
 }
```

**Comment by chalasr** [comment]
FYI the integration of the Templating component in FrameworkBundle is deprecated in Symfony 4.3 https://github.com/symfony/symfony/pull/21035 (i.e. `framework.templating` configuration + `templating.*` services are deprecated)

**Comment by gilles-g** [comment]
Thx for you help, I have try a lot of things but not that.. :thinking: 

If I use use `Symfony\Component\Templating\EngineInterface` is ok, right?

**Comment by chalasr** [comment]
Nope, using the interface itself is fine, but you're relying on the `templating` service which is provided by FrameworkBundle (that's what is deprecated) in https://github.com/lexik/LexikCronFileGeneratorBundle/blob/master/Resources/config/dump_file.xml#L13. The bundle should use Twig directly instead, don't hesitate to ping me

File: composer.json [hunk 1]
```diff
@@ -13,9 +13,7 @@
     "require": {
         "php": "^7.0",
         "symfony/console": "^3.4|^4.0",
-        "symfony/framework-bundle":  "^3.4|^4.0",
-        "symfony/templating": "^3.4|^4.0",
-        "symfony/twig-bundle": "^3.4|^4.0"
```
Comment (chalasr) [inline composer.json]: framework-bundle and `twig-bundle` should probably be kept (as they provide dependency-injection and http-kernel, required for a bundle)

Comment (gilles-g) [reply composer.json]: Yeah that's was too much :dagger:

**Comment by gilles-g** [comment]
@chalasr Green is the new red! Thx for you help ;)

**Pull request merged**